src/reactpy/backend/_common.py

```diff
@@ -31,29 +30,32 @@ async def serve_development_asgi(
     port: int,
     started: asyncio.Event | None,
 ) -> None:
-    """Run a development server for starlette"""
-    server = UvicornServer(
-        UvicornConfig(
+    """Run a development server for an ASGI application"""
+    server = uvicorn.Server(
+        uvicorn.Config(
             app,
             host=host,
             port=port,
             loop="asyncio",
             reload=True,
         )
     )
-
+    server.config.setup_event_loop()
     coros: list[Awaitable[Any]] = [server.serve()]
 
+    # If a started event is provided, then use it signal based on `server.started`
     if started:
         coros.append(_check_if_started(server, started))
 
     try:
         await asyncio.gather(*coros)
     finally:
+        if not hasattr(server, "servers"):  # pragma: no cover
+            server.servers = []
```

Comment:
What's going on with this line here?

Reply:
Because we aren't using the uvicorn's `run()` API, we can't guarantee uvicorn's order of operations. So we need to make sure ``shutdown()` always has an initialized list of `self.servers` to use.

See [this issue](https://github.com/encode/uvicorn/pull/1845) for more details.